CMakeLists.txt

```diff
@@ -53,6 +53,15 @@ message(STATUS "UMD build type: ${CMAKE_BUILD_TYPE}")
 
 include(dependencies)
 
+# Enable all warnings and treat them as errors
+add_compile_options(
+    -Wall
+    -Werror
+    -Wno-unused-variable
+    -Wno-unused-function
+    -Wno-unused-but-set-variable
+)
```

Comment:
Are these flags the same between clang and gcc?

Reply:
I've added GCC CI build in #384, so that one should hopefully be enough to verify this.